src/editor/deserialize.js

```diff
@@ -16,73 +16,83 @@ limitations under the License.
 */
 
 import { MATRIXTO_URL_PATTERN } from '../linkify-matrix';
-import { PlainPart, UserPillPart, RoomPillPart, NewlinePart } from "./parts";
 import { walkDOMDepthFirst } from "./dom";
 
 const REGEX_MATRIXTO = new RegExp(MATRIXTO_URL_PATTERN);
 
-function parseLink(a, room, client) {
+function parseAtRoomMentions(text, partCreator) {
+    const ATROOM = "@room";
+    const parts = [];
+    text.split(ATROOM).forEach((textPart, i, arr) => {
+        if (textPart.length) {
+            parts.push(partCreator.plain(textPart));
+        }
+        const isLast = i === arr.length - 1;
```

Comment:
This variable feels a bit misleading, as I was about to comment that it seems to filter out `@room` pills in the middle of the text. However, some black magic causes them to work just fine... somehow?

A comment to explain might also help.

ftr I was testing with this text: `"@room check out the @room changes I made to the @room"`

Reply:
We split the string with `"@room"` as a separator, so in the above case, we'd get the following array back:
```
[ '', ' check out the ', ' changes I made to the ', '' ]
```
E.g. an empty string before the first, and after the last encounter of the separator. So no need to append a pill for the last match.

Wonder how to best make this clear ... I added a comment, but feel free to suggest any naming if you'd prefer.

Reply:
ah ha, the comment does help. Thank you!